src/components/Wallet/WalletSpend.jsx

```diff
@@ -31,41 +32,43 @@ import {
   SPEND_STEP_PREVIEW,
   SPEND_STEP_SIGN,
   setSpendStep as setSpendStepAction,
+  deleteChangeOutput as deleteChangeOutputAction,
 } from "../../actions/transactionActions";
+import { naiveCoinSelection } from "../../utils";
 
 // Components
 import NodeSet from "./NodeSet";
 import OutputsForm from "../ScriptExplorer/OutputsForm";
 import WalletSign from "./WalletSign";
 import TransactionPreview from "./TransactionPreview";
+import { bigNumberPropTypes } from "../../proptypes/utils";
 
 class WalletSpend extends React.Component {
   outputsAmount = new BigNumber(0);
 
+  coinSelection = naiveCoinSelection;
+
   feeAmount = new BigNumber(0);
 
-  componentDidUpdate() {
-    const { autoSpend, finalizedOutputs } = this.props;
-    if (autoSpend && !finalizedOutputs) {
-      setTimeout(this.selectCoins, 0);
+  componentDidUpdate = (prevProps) => {
+    const { finalizedOutputs } = this.props;
+    if (finalizedOutputs && !prevProps.finalizedOutputs) {
```

Comment:
If the state went from the outputs not being finalized to they are finalized, then we assume the preview view is ready to be shown.

Reply:
Not super happy about this way to get to the preview view since it's making some assumptions about the state and expecting that certain actions are taken elsewhere (i.e. that `finalizedOutputs` is set to true by the coin selection action creator),  but it gets the job done.